### ISSUE TITLE
[Bug] Usage of Head component results in ‘Maximum call stack exceeded’

### ISSUE THREAD

**Issue by Cinamonas** [nozzle/react-static#1119]
### Environment

```
  System:
    OS: macOS 10.14.4
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Memory: 18.74 MB / 16.00 GB
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 8.11.3 - ~/.nvm/versions/node/v8.11.3/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v8.11.3/bin/npm
  Browsers:
    Chrome: 73.0.3683.103
  npmPackages:
    react: ^16.8.2 => 16.8.6 
    react-dom: ^16.8.2 => 16.8.6 
    react-static: ^7.0.7 => 7.0.7 
    react-static-plugin-reach-router: ^7.0.0 => 7.0.0 
    react-static-plugin-scss-modules: ^1.0.3 => 1.0.3 
    react-static-plugin-sitemap: ^7.0.0 => 7.0.0 
    react-static-plugin-source-filesystem: ^7.0.0 => 7.0.0 
```

### Steps to Reproduce the problem

1. `react-static create --name test --template basic`
2.  Edit `about.js` by adding the following:
```diff
  // about.js
  import React from 'react'
+ import { Head } from 'react-static'

  export default () => (
    <div>
+     <Head>
+       <title>Test</title>
+     </Head>
      <p>React Static is a progressive static site generator for React.</p>
    </div>
  )
```
3. `yarn start`
4. Open the app in the browser
5. Click on _About_ in the nav
6. The _About_ page is opened, title updated to _Test_
7. Click on _About_ **again**
6. Behold the `RangeError: Maximum call stack size exceeded`

### Expected Behavior

No error.

**Comment by Cinamonas** [comment]
I did some digging and found that this is React Helmet’s issue: https://github.com/nfl/react-helmet/issues/373. Allegedly fixed in v6.0.0-beta, but it hasn’t moved from beta since.

A workaround is to use props instead of children:
```diff
<Head
+  title="Test"
+  meta={[
+    {
+      name: 'description',
+      content: 'A description goes here…',
+    },
+  ]}
>
-  <title>About</title>
-  <meta name="description" content="A description goes here…" />
</Head>
```

Interestingly, following the exact same steps to reproduce with React-Static v6, doesn’t result in this issue. And I didn’t have this issue with v5. So it means that some changes in v7 surface it. Will need to keep digging ⛏️

**Comment by kevinmanncito** [comment]
I've got a potential fix for this. React static should just need to bump the version of react-helmet that is being used from `5.2.0` to `5.2.1`. For now I just installed react-helmet myself and the issue went away. Looks like if react static bumps the version then that would fix it.

**Comment by tannerlinsley** [comment]
Please submit a PR :)

**Comment by bard** [comment]
At least if using yarn, running `yarn upgrade` is enough, since react-static's package.json states a dependency on react-helmet 5.2.0 or higher.

**Comment by kosich** [comment]
Same with npm. I don't recall exact steps, yet removing `package-lock.json` with `npm install` should help.

`react-helmet` fellas fixed it in `5.2.1` by switching from `deep-equal` dependency to `react-fast-compare`. [See this commit](https://github.com/nfl/react-helmet/commit/1803f678b9d38b5ae91961ea13ea1c7fa7c9afe8).

 @Cinamonas, please, verify and close this issue, if it works fine now.

_should not be related, though I also used [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) to ensure I'm on latest versions of my direct dependencies_

**Comment by Cinamonas** [comment]
Huh, I somehow missed that not v6.0.0-beta, but v5.2.1 fixes the issue. I can confirm that it does indeed work after bumping to v5.2.1 and created a PR for that.

**Comment by kirklikecaptain** [comment]
Any updates on this issue?

**Comment by Cinamonas** [comment]
@kirklikecaptain, this issue will be automatically closed once #1175 is merged. But until then, simply running `yarn upgrade` should do the trick.
